### PR TITLE
Replace circleci with github actions and update dependencies

### PR DESCRIPTION
**Pull request by bayounetta**
Windows support is still a work in progress but I'm not sure if we need it since we didn't have it before

### CODE REVIEW

**Comment by bayounetta** [comment]
Removed windows test runs but left windows cache builds since those are already working.
I've also gone ahead and changed the branch protection rules to be based on Github Actions rather than CircleCI.

Opting to make windows test runs and code coverage comments on pull requests into their own issues.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
           cache: "poetry"
       - name: lint codebase
-        run: poetry run flake8 --ignore=E501,E402 .
+        run: poetry run flake8 --ignore=E501,E402,E275 .
```
Comment (tomrittervg) [inline .github/workflows/ci.yml:45]: This commit message should say what E275 is and why we ignore it

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      fail-fast: false
+      matrix:
+        python-version: ["3.7"]
+        poetry-version: [1.1.0]
+        os: [ubuntu-18.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
```
Comment (tomrittervg) [inline .github/workflows/ci.yml:39]: Do we still need to checkout and install pip and setup python here (and below) if it's in the build task we depend on?

**Review by tomrittervg — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (bayounetta) [reply .github/workflows/ci.yml:39]: The caching from the build step is for dependencies only, sadly, so we do still need to do that. each of the jobs runs in its own container using either github's `ubuntu-latest` or `windows-latest` image.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bayounetta) [reply .github/workflows/ci.yml:45]: commit message updated!

**Review by tomrittervg — APPROVED** [review]

**Pull request merged**